Difi triggering FileExistsError: [Errno 17] File exists: '/DIFI_ARRAY' error

Today I found quite suddenly that any calls to difi resulted in a `FileExistsError: [Errno 17] File exists: '/DIFI_ARRAY' `error which persisted even in a fresh environment. (I'll post the full traceback below.)

Installing v.2rc3 stopped this as now shared memory arrays have an instance-specific name including the PID. So I think it's a good idea to update the requirement for difi to this version in the pyproject.toml file.

I'm not sure why this happened (looks like a file isn't being cleared from memory when it should be?) but I assume that if it happened to me it could happen to someone else.

```
tests/sorcha/test_integration.py:24: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
src/sorcha/sorcha.py:234: in runLSSTPostProcessing
    observations = PPLinkingFilter(
src/sorcha/modules/PPLinkingFilter.py:85: in PPLinkingFilter
    findable, _ = metric.run(
/opt/miniconda3/envs/sorcha/lib/python3.9/site-packages/difi/metrics.py:1063: in run
    findable = self.run_by_object(
/opt/miniconda3/envs/sorcha/lib/python3.9/site-packages/difi/metrics.py:742: in run_by_object
    self._store_as_shared_record_array(objects, obs_ids, times, ra, dec, nights)
/opt/miniconda3/envs/sorcha/lib/python3.9/site-packages/difi/metrics.py:559: in _store_as_shared_record_array
    shared_mem = shared_memory.SharedMemory("DIFI_ARRAY", create=True, size=observations_array.nbytes)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <[AttributeError("'NoneType' object has no attribute 'startswith'") raised in repr()] SharedMemory object at 0x191162d60>, name = '/DIFI_ARRAY'
create = True, size = 21140

    def __init__(self, name=None, create=False, size=0):
        if not size >= 0:
            raise ValueError("'size' must be a positive integer")
        if create:
            self._flags = _O_CREX | os.O_RDWR
            if size == 0:
                raise ValueError("'size' must be a positive number different from zero")
        if name is None and not self._flags & os.O_EXCL:
            raise ValueError("'name' can only be None if create=True")
    
        if _USE_POSIX:
    
            # POSIX Shared Memory
    
            if name is None:
                while True:
                    name = _make_filename()
                    try:
                        self._fd = _posixshmem.shm_open(
                            name,
                            self._flags,
                            mode=self._mode
                        )
                    except FileExistsError:
                        continue
                    self._name = name
                    break
            else:
                name = "/" + name if self._prepend_leading_slash else name
>               self._fd = _posixshmem.shm_open(
                    name,
                    self._flags,
                    mode=self._mode
                )
E               FileExistsError: [Errno 17] File exists: '/DIFI_ARRAY'

/opt/miniconda3/envs/sorcha/lib/python3.9/multiprocessing/shared_memory.py:103: FileExistsError
```

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -28,7 +28,7 @@ dependencies = [
     "matplotlib",
     "sbpy",
     "tables",
-    "difi == 1.2rc2",
+    "difi == 1.2rc3",
 ]
 
 [project.scripts]

